### PR TITLE
Add support for listing available SObjects

### PR DESCRIPTION
**Pull request by nyergler**

### CODE REVIEW

File: sobject/list.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		var respErrs []sfdc.Error
+		err = decoder.Decode(&respErrs)
+		var errMsg error
+		if err == nil {
+			for _, respErr := range respErrs {
+				errMsg = fmt.Errorf("list response err: %s: %s", respErr.ErrorCode, respErr.Message)
+			}
+		} else {
+			errMsg = fmt.Errorf("list response err: %d %s", response.StatusCode, response.Status)
+		}
```
Comment (vtopc) [inline sobject/list.go:71]: Hi. We are currently on half way of switching from `[]sfdc.Error` to `sfdc.HandleError()`.
Not saying to replace all these code to the `sfdc.HandleError()` in this PR, but we might do that later for consistency.

File: sobject/list.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+type list struct {
+	session session.ServiceFormatter
+}
+
+func (l *list) callout() (ListValue, error) {
+
+	request, err := l.request()
+
```
Comment (vtopc) [inline sobject/list.go:24]: I know that this was a copy-paste from another file, but please, remove these empty lines before error checks.

File: sobject/list.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"net/http"
+
+	"github.com/namely/go-sfdc/v3"
+	"github.com/namely/go-sfdc/v3/session"
+)
+
+// ListValue is a structure that is returned from Salesforce SObject list calls
+type ListValue struct {
+	SObjects []DescribeValue `json:"sobjects"`
+}
```
Comment (vtopc) [inline sobject/list.go:15]: Sorry, I don't know SF API well. Isn't this the same as a query (`soql.NewQuery()`)?

File: sobject/list.go (hunk 1 above)
Comment (nyergler) [reply sobject/list.go:71]: Happy to do so! Can you point me to an example of the new preferred format?

File: sobject/list.go (hunk 3 above)
Comment (nyergler) [reply sobject/list.go:15]: I don't _think_ they're the same -- a query (`soql.NewQuery`) is a free form query and can be used to fetch or mutate records; the List API doesn't return actual records, it only describes the available record _types_. For our application we're interested in determining what types -- built-in and custom -- the user's SF org has (Salesforce API docs [here](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_describeGlobal.htm). 

_And_, I'm not an expert, either :).

**Comment by nyergler** [comment]
Thanks for the review, @vtopc; let me know if you have additional questions.

File: sobject/list.go (hunk 3 above)
Comment (vtopc) [reply sobject/list.go:14]: Got it. Thanks for the docs.

**Comment by vtopc** [comment]
@nyergler, all good, just fix https://github.com/namely/go-sfdc/pull/11/files#r463701887

File: sobject/list.go (hunk 2 above)
Comment (nyergler) [reply sobject/list.go]: Done, thanks for pointing that out.

**Pull request merged**